### PR TITLE
File system usage refactor

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
As part of the builder moving to be permanently in the Cloud instance (self hosted or sandbox) we need to do away with any real usage of the file system.

There are two exceptions to this rule:
1. Temporary files, occasionally temporary files will be used for say, downloading a tarball and extracting it, a multi-part file upload to Koa will be stored in temp files and so on - as long as the system only depends on the file being present in the filesystem for the duration of an API call, it is allowable.
2. Packages, if the server needs to use a specific package then it will need to download it to the filesystem. In theory we could download packages to minio and download them to a temporary file each time we want to run them but it is just easier to make sure our versioning is correct and allow each server instance to fetch its own copy of that version of the package - an example of this is external automation steps.

All other file system access had to be removed as part of this update, the main offender being the `~/.budibase` directory where pouchDB info was stored, attachments and client lib pre upload to the object store. Now with this update all these files will be sent directly to the object store.

In this update I also included some changes to the single stack development experience, simply so that the environment gets seeded with good development env variables to point at the dev cluster automatically.

To perform this update I built out `server/src/utilities/fileSystem` a set of utilities that "act" like file system access, performing specific tasks which we previously performed on the file system. Under the hood this depends on a mixture of temporary files and the object store client to make sure all info is persisted correctly.

Lastly I also iterated through the environment variables to remove the previous `CLOUD` env variable that a lot of the system depended on to determine if an operation was running in the Cloud or in the builder - now this is not of importance. I have changed some of these features to always run as we are always running in the cloud now and other features were hidden behind development/production checks via `env.isDev()`, `env.isProd()` and `env.isTest()`.

### CODE REVIEW

**Comment by kevmodrome** [comment]
What a beast of a PR. Good job!

File: packages/server/scripts/exportAppTemplate.js [hunk 1]
```diff
@@ -22,18 +25,22 @@ yargs
...
         return
       }
-      const exportPath = await exportTemplateFromApp({
-        templateName: args.name,
-        appId: args.appId,
-      })
-      console.log(`Template ${args.name} exported to ${exportPath}`)
+      const exportPath = join(budibaseAppsDir(), "templates", "app", name, "db")
+      fs.ensureDirSync(exportPath)
+      const writeStream = fs.createWriteStream(join(exportPath, "dump.text"))
```
Comment (aptkingston) [inline packages/server/scripts/exportAppTemplate.js:39]: Is the budibase home directory always the one used for these temporary file ops? Cause I know there's a budibase temp dir too. Just wondering when running the server in dev now (outside docker), what are the actual file paths that are used. Are the home dir and temp dir both still used?

**Review by aptkingston — APPROVED** [review]
Huge! 🎉 That was a chonky piece of work, super job.

File: packages/server/scripts/exportAppTemplate.js (hunk 1 above)
Comment (mike12345567) [reply packages/server/scripts/exportAppTemplate.js:39]: That old script doesn't really have a purpose anymore, to be honest its not entirely possible to script it that way, realistically need to rewrite that script - although you can export from an endpoint on the server. I might actually just remove the script cause its just not useful anymore.

On your question, those directories outside docker can never be hit, we don't actually use the apps directory even in development with the server running externally, everything that happens happens in the OS system level temporary files directory (unix `/tmp`). The home directory is still technically used to download automations just so they live on somewhere between reboots etc and aren't wiped out in anyway shape or form but apart from that all files which get created should exist in the `/tmp/.budibase` directory.

**Pull request merged**